osu.Game/Screens/Play/HUD/DefaultPerformancePointsCounter.cs

```diff
@@ -0,0 +1,159 @@
+            {
+                gameplayRuleset = player.GameplayRuleset;
+                timedAttributes = gameplayRuleset.CreateDifficultyCalculator(new GameplayWorkingBeatmap(player.GameplayBeatmap)).CalculateTimed(player.Mods.Value.ToArray()).ToArray();
+            }
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            if (scoreProcessor != null)
+                scoreProcessor.NewJudgement += onNewJudgement;
+        }
+
+        private void onNewJudgement(JudgementResult judgement)
+        {
+            if (player == null || timedAttributes.Length == 0)
+                return;
+
+            var attribIndex = Array.BinarySearch(timedAttributes, 0, timedAttributes.Length, new TimedDifficultyAttributes(judgement.HitObject.GetEndTime(), null));
+            if (attribIndex < 0)
+                attribIndex = ~attribIndex - 1;
+            attribIndex = Math.Clamp(attribIndex, 0, timedAttributes.Length - 1);
+
+            var ppProcessor = gameplayRuleset.CreatePerformanceCalculator(timedAttributes[attribIndex].Attributes, player.Score.ScoreInfo);
+            Current.Value = (int)(ppProcessor?.Calculate() ?? 0);
+        }
+
+        protected override LocalisableString FormatCount(int count) => count.ToString(@"D");
+
+        protected override IHasText CreateText() => new TextComponent();
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (scoreProcessor != null)
+                scoreProcessor.NewJudgement -= onNewJudgement;
+        }
+
+        private class TextComponent : CompositeDrawable, IHasText
+        {
+            public LocalisableString Text
+            {
+                get => text.Text;
+                set => text.Text = value;
+            }
+
+            private readonly OsuSpriteText text;
+
+            public TextComponent()
+            {
+                AutoSizeAxes = Axes.Both;
+
+                InternalChild = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Spacing = new Vector2(2),
+                    Children = new Drawable[]
+                    {
+                        text = new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Font = OsuFont.Numeric.With(size: 16)
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Text = @"pp",
+                            Font = OsuFont.Numeric.With(size: 8)
+                        }
+                    }
+                };
+            }
+        }
+
+        private class GameplayWorkingBeatmap : WorkingBeatmap
```

Comment:
maybe this is okay, but needs a TODO mentioning that it should be replaced and not a thing going forward.